typescript/lib/config.ts

```diff
@@ -233,12 +233,40 @@ export class AIConfigRuntime implements AIConfig {
    * Saves this AIConfig to a file.
    * @param filePath The path to the file to save to.
    * @param saveOptions Options that determine how to save the AIConfig to the file.
+   * @param mode Whether to save the AIConfig as JSON or YAML. If unspecified, the file extension will be used to determine the mode.
    */
   public save(
     filePath?: string,
     saveOptions?: SaveOptions,
     mode?: "json" | "yaml"
   ) {
+    const defaultFilePath = mode === "yaml" ? "aiconfig.yaml" : "aiconfig.json";
+    if (!filePath) {
+      filePath = this.filePath ?? defaultFilePath;
+    }
+
+    if (mode == null) {
+      if (isYamlExt(filePath)) {
+        mode = "yaml";
+      } else {
+        // Default to JSON
+        mode = "json";
+      }
+    }
+
+    const aiConfigString = this.toString(saveOptions, mode);
+    fs.writeFileSync(filePath, aiConfigString);
+  }
+
+  /**
+   * Returns this AIConfig as a string.
+   *
+   * Note that this doesn't return the full string representation of the AIConfig object,
+   * but rather the string representation of the AIConfig object that can be persisted to a file.
```

Comment:
nit, "string representation" -> "serialized representation" and also probably worth explaining what the difference between these is